src/main/java/net/dv8tion/jda/core/entities/MessageActivity.java

```diff
@@ -0,0 +1,216 @@
+            this.coverId = coverId;
+            this.id = id;
+        }
+
+        /**
+         * The name of this {@link net.dv8tion.jda.core.entities.MessageActivity.Application}.
+         *
+         * @return the applications name.
+         */
+        public String getName()
+        {
+            return name;
+        }
+
+        /**
+         * A short description of this {@link net.dv8tion.jda.core.entities.MessageActivity.Application}.
+         *
+         * @return the applications description.
+         */
+        public String getDescription()
+        {
+            return description;
+        }
+
+        /**
+         * The icon id of this {@link net.dv8tion.jda.core.entities.MessageActivity.Application}.
+         *
+         * @return the applications icon id.
+         */
+        public String getIconId()
+        {
+            return iconId;
+        }
+
+        /**
+         * The url of the icon image for this application.
+         *
+         * @return the url of the icon
+         */
+        public String getIconUrl()
+        {
+            return "https://cdn.discordapp.com/application/" + getId() + "/" + iconId + ".png";
+        }
+
+        /**
+         * The cover aka splash id of this {@link net.dv8tion.jda.core.entities.MessageActivity.Application}.
+         *
+         * @return the applications cover image/id.
+         */
+        public String getCoverId()
+        {
+            return coverId;
+        }
+
+        /**
+         * The url of the cover image for this application.
+         *
+         * @return the url of the cover/splash
+         */
+        public String getCoverUrl()
+        {
+            return "https://cdn.discordapp.com/application/" + getId() + "/" + coverId + ".png";
+        }
+
+        @Override
+        public long getIdLong()
+        {
+            return id;
+        }
+    }
+
+    /**
+     * An enum containing {@link net.dv8tion.jda.core.entities.MessageActivity MessageActivity} types.
+     */
+    public enum ActivityType
+    {
+        /**
+         * The {@link net.dv8tion.jda.core.entities.MessageActivity MessageActivity} type used for inviting people to join a game.
+         */
+        JOIN(1),
+        /**
+         * The {@link net.dv8tion.jda.core.entities.MessageActivity MessageActivity} type used for inviting people to spectate a game.
+         */
+        SPECTATE(2),
+        /**
+         * The {@link net.dv8tion.jda.core.entities.MessageActivity MessageActivity} type used for inviting people to listen (Spotify) together.
+         */
+        LISTENING(3),
+        /**
+         * The {@link net.dv8tion.jda.core.entities.MessageActivity MessageActivity} type used for requesting to join a game.
+         */
+        JOIN_REQUEST(5),
+        /**
+         * Unknown Discord {@link net.dv8tion.jda.core.entities.MessageActivity MessageActivity} type. Should never happen and would only possibly happen if Discord implemented a new
```

Comment:
"Should never and would only" this sounds rather confusing, just say that this represents all unhandled types